### DISCUSSION TITLE
Doubts on Sáida Authentication

### DISCUSSION THREAD

**Discussion by ooclaar** [haraka/Haraka#3053] [Q&A] [answered]
I am using a service called post.io that uses Haraka to send the messages. I wonder if anyone can help me set up an alternate output using AWS or any other SMTP when a message has its delivery rejected by a server.

**Comment by msimerson** [comment, marked as answer]
I'm pretty sure this can be done out-of-the-box. For incoming emails, the smtp_forward plugin has per-domain routing that I aded a long time ago. It lets you send incoming emails to particular backend servers. For outbound, you can use the `get_mx` hook and add custom routes for individual domains.

**Reply by ooclaar** [reply]
Could you tell me where to start? I have basic knowledge of Javascript programming. But I would like to understand and master it better to make my own customizations.

**Comment by msimerson** [comment]
You start with our [plugin docs](https://github.com/haraka/Haraka/blob/master/docs/Plugins.md) which let you plug into various parts of the SMTP conversations.
